matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/query/ReadQueries.kt

```diff
@@ -59,6 +59,25 @@ internal fun isEventRead(realmConfiguration: RealmConfiguration,
     return isEventRead
 }
 
+private fun handleMissingEvent(realm: Realm, latestChunkEntity: ChunkEntity, roomId: String, userId: String, eventId: String): Boolean {
+    return if (realm.doesEventExistInChunkHistory(eventId) && realm.hasReadReceiptInLatestChunk(latestChunkEntity, roomId, userId)) {
+        MARK_OLD_EVENT_AS_READ
+    } else {
+        // This can happen when fast lane events are displayed before the database finishes updating
+        MARK_UNREAD_DUE_TO_FASTLANE
+    }
+}
+
+private fun Realm.doesEventExistInChunkHistory(eventId: String): Boolean {
+    return ChunkEntity.findIncludingEvent(this, eventId) != null
+}
+
+private fun Realm.hasReadReceiptInLatestChunk(latestChunkEntity: ChunkEntity, roomId: String, userId: String) : Boolean {
+    return ReadReceiptEntity.where(this, roomId, userId).findFirst()?.let {
```

Comment:
Nit: The code is OK, but maybe use named arguments for roomId and userId to avoid confusion, since the types are the same (String)

Reply:
:100: will do